src/gait/ai_commit.py

```diff
@@ -0,0 +1,96 @@
+import os
+from openai import OpenAI
+from dotenv import load_dotenv
+import subprocess
+
+def get_git_diff():
+    """Get the staged changes diff"""
+    try:
+        result = subprocess.run(
+            ["git", "diff", "--staged"],
+            capture_output=True,
+            text=True,
+            check=True
+        )
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        print(f"Error getting git diff: {e}")
+        return None
+
+def handle_ai_commit():
+    diff_text = get_git_diff()
+    if not diff_text:
+        print("No staged changes found.")
+        return 1
+    
+    commit_message = generate_commit_message(diff_text)
+    if not commit_message:
+        return 1
+    
+    # Show the commit message and ask for confirmation
+    print("\n--- Proposed commit message ---")
+    print(f"└─→ {commit_message}")  # Bold green arrow and message
+    while True:
+        response = input("\n\033[0;36mDo you want to proceed with this commit message? (y[es]/n[o]/e[dit]): \033[0m").lower().strip()
+        if response in ['y', 'yes']:  # Empty response defaults to yes
+            break
+        elif response in ['n', 'no']:
+            print("\033[0;31mCommit cancelled.\033[0m")  # Bold red for cancellation
+            return 1
+        elif response in ['e', 'edit']:
+            new_message = input("New commit message: ").strip()
+            if new_message:
+                commit_message = new_message
+                break
+            print("Invalid message. Please try again.")
+        else:
+            print("Please answer 'y' (yes), 'n' (no), or 'e' (edit)")
+    
+    try:
+        subprocess.run(["git", "commit", "-m", commit_message], check=True)
+        return 0
+    except subprocess.CalledProcessError as e:
+        print(f"Error creating commit: {e}")
+        return e.returncode
+
+def generate_commit_message(diff_text):
+    """Generate commit message using OpenAI API"""
+    load_dotenv()
+    api_key = os.getenv("OPENAI_API_KEY")
+    if not api_key:
+        print("Error: OPENAI_API_KEY not found in environment variables. Please set it in .env file.")
+        return None
+    
+    client = OpenAI(api_key=api_key)
+    
+    user_prompt = f"""The following Git diff input is in Unified Diff format, 
```

Comment:
Have you had any issues with the model returning additional output? 

we can try leading the prompt. So we can add something like this to the end of it. 

Git diff: {diff_text}

Generated Diff: 


This should suggest to the model that it should only complete the prompt instead of trying to have a conversation